File: rb/features.rb
# Bng2latlongConverter SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module Bng2latlongConverterFeatures
  def self.make_feature(name)
    case name
    when "base"
      Bng2latlongConverterBaseFeature.new
    when "ratelimit"
      Bng2latlongConverterRatelimitFeature.new
    when "retry"
      Bng2latlongConverterRetryFeature.new
    when "test"
      Bng2latlongConverterTestFeature.new
    when "timeout"
      Bng2latlongConverterTimeoutFeature.new
    else
      Bng2latlongConverterBaseFeature.new
    end
  end
end

File: py/features.py
# Bng2latlongConverter SDK feature factory

from feature.base_feature import Bng2latlongConverterBaseFeature
from feature.test_feature import Bng2latlongConverterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: Bng2latlongConverterBaseFeature(),
        "test": lambda: Bng2latlongConverterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
